plugins/reach/lib/kReachManager.php

```diff
@@ -123,19 +122,24 @@ public function consumeEvent(KalturaEvent $event)
 		$scope = $event->getScope();
 		$partnerId = $scope->getPartnerId();
 		$object = $scope->getObject();
-		$entryId = $object->getEntryId();
 		foreach (self::$booleanNotificationTemplatesFulfilled as $booleanNotificationTemplate)
 		{
 			$profileId = $booleanNotificationTemplate[self::PROFILE_ID];
 			$fullFieldCatalogItemIds = $booleanNotificationTemplate[self::ACTION][0]->getCatalogItemIds();
+			$taskObjectType = $booleanNotificationTemplate[self::ACTION][0]->getEntryObjectType();
 			$fullFieldCatalogItemIdsArr = array_map('trim', explode(',', $fullFieldCatalogItemIds));
 			$allowedCatalogItemIds = PartnerCatalogItemPeer::retrieveActiveCatalogItemIds($fullFieldCatalogItemIdsArr, $partnerId);
 			if(!count($allowedCatalogItemIds))
 			{
 				KalturaLog::debug('None of the fulfilled catalog item ids are active on partner, [' . implode(',', $fullFieldCatalogItemIds) . ']');
 				continue;
 			}
-			$this->addingEntryVendorTaskByObjectIds($entryId, $allowedCatalogItemIds, $profileId, $object);
+			$vendorTaskObjectHandler = HandlerFactory::getHandlerAutomaticFlow($taskObjectType, $object);
```

Comment:
since this is a duplicate code 
$vendorTaskObjectHandler = HandlerFactory::getHandlerAutomaticFlow($taskObjectType, $object);
					if(!$vendorTaskObjectHandler)
					{
						continue;
					}
					
					let's move it inside addingEntryVendorTaskByObjectIds and add relevant log msg if vendorTaskObjectHandler is null